great_expectations/rule_based_profiler/domain_builder/cardinality_checker.py

```diff
@@ -0,0 +1,211 @@
+    ABS_10 = AbsoluteCardinalityLimit("ABS_10", 10)
+    ABS_100 = AbsoluteCardinalityLimit("ABS_100", 100)
+    ABS_1000 = AbsoluteCardinalityLimit("ABS_1000", 1000)
+    ABS_10_000 = AbsoluteCardinalityLimit("ABS_10_000", int(1e4))
+    ABS_100_000 = AbsoluteCardinalityLimit("ABS_100_000", int(1e5))
+    ABS_1_000_000 = AbsoluteCardinalityLimit("ABS_1_000_000", int(1e6))
+    ABS_10_000_000 = AbsoluteCardinalityLimit("ABS_10_000_000", int(1e7))
+    ABS_100_000_000 = AbsoluteCardinalityLimit("ABS_100_000_000", int(1e8))
+    ABS_1_000_000_000 = AbsoluteCardinalityLimit("ABS_1_000_000_000", int(1e9))
+    REL_0 = RelativeCardinalityLimit("REL_0", 0.0)
+    REL_001 = RelativeCardinalityLimit("REL_001", 1e-5)
+    REL_01 = RelativeCardinalityLimit("REL_01", 1e-4)
+    REL_0_1 = RelativeCardinalityLimit("REL_0_1", 1e-3)
+    REL_1 = RelativeCardinalityLimit("REL_1", 1e-2)
+    REL_10 = RelativeCardinalityLimit("REL_10", 0.10)
+    REL_25 = RelativeCardinalityLimit("REL_25", 0.25)
+    REL_50 = RelativeCardinalityLimit("REL_50", 0.50)
+    REL_75 = RelativeCardinalityLimit("REL_75", 0.75)
+    ONE_PCT = RelativeCardinalityLimit("ONE_PCT", 0.01)
+    TEN_PCT = RelativeCardinalityLimit("TEN_PCT", 0.10)
+
+
+class CardinalityChecker:
+    """Handles cardinality checking given cardinality limit mode and measured value.
+
+    This class also validates cardinality limit settings and converts from
+    various types of settings. You can choose one of the attributes listed
+    below to create an instance.
+
+    Attributes:
+        limit_mode: CardinalityLimitMode or string name of the mode
+            defining the maximum allowable cardinality.
+        max_unique_values: number of max unique rows for a custom
+            cardinality limit.
+        max_proportion_unique: proportion of unique values for a
+            custom cardinality limit.
+    """
+
+    SUPPORTED_CARDINALITY_LIMIT_MODE_CLASSES: Tuple[
+        Union[AbsoluteCardinalityLimit, RelativeCardinalityLimit]
+    ] = (
+        AbsoluteCardinalityLimit,
+        RelativeCardinalityLimit,
+    )
+    SUPPORTED_CARDINALITY_LIMIT_MODE_STRINGS: Tuple[str] = (
+        mode.name for mode in CardinalityLimitMode
+    )
+
+    def __init__(
+        self,
+        limit_mode: Optional[Union[CardinalityLimitMode, str]] = None,
+        max_unique_values: Optional[int] = None,
+        max_proportion_unique: Optional[Union[int, float]] = None,
+    ):
+        self.supported_limit_mode_class_names = (
+            mode.__name__ for mode in self.SUPPORTED_CARDINALITY_LIMIT_MODE_CLASSES
+        )
+
+        self._limit_mode = self._convert_to_cardinality_mode(
+            limit_mode=limit_mode,
+            max_unique_values=max_unique_values,
+            max_proportion_unique=max_proportion_unique,
+        )
+
+    @property
+    def limit_mode(self) -> CardinalityLimitMode:
+        return self._limit_mode
+
+    def cardinality_within_limit(self, metric_value: Union[int, float]) -> bool:
```

Comment:
@anthonyburdi how about using `Number` type instead of `Union[int, float]` -- might it work?  Thanks!

Reply:
@alexsherstinsky good call, but looks like the recommendation is to use `float` instead: https://www.python.org/dev/peps/pep-0484/#the-numeric-tower

Reply:
Changed in a local commit, will push after resolving other comments.